parser/src/main/scala/hmda/parser/fi/lar/LarCsvParser.scala

```diff
@@ -83,63 +87,67 @@ object LarCsvParser {
         )
       val denial = Denial(denial1, denial2, denial3)
 
-      Right(LoanApplicationRegister(
-        id,
-        respId,
-        agencyCode,
-        loan,
-        preapprovals,
-        actionType,
-        actionDate,
-        geography,
-        applicant,
-        purchaserType,
-        denial,
-        rateSpread,
-        hoepaStatus,
-        lienStatus
-      ))
+      Right(
+        LoanApplicationRegister(
+          id,
+          respId,
+          agencyCode,
+          loan,
+          preapprovals,
+          actionType,
+          actionDate,
+          geography,
+          applicant,
+          purchaserType,
+          denial,
+          rateSpread,
+          hoepaStatus,
+          lienStatus
+        )
+      )
     } else {
-      Left(larErrors)
+      val lErrors = validation.toEither.left.get
+      Left(lErrors.list.toList)
     }
   }
 
-  def checkLar(s: String): List[String] = {
-    val values = s.split('|').map(_.trim)
-    var errors = List[String]()
-    val numericFields = Map(
-      0 -> "Record Identifier",
-      2 -> "Agency Code",
-      5 -> "Loan Type",
-      6 -> "Property Type",
-      7 -> "Loan Purpose",
-      8 -> "Owner Occupancy",
-      9 -> "Loan Amount",
-      10 -> "Preapprovals",
-      11 -> "Type of Action Taken",
-      12 -> "Date of Action",
-      17 -> "Applicant Ethnicity",
-      18 -> "Co-applicant Ethnicity",
-      19 -> "Applicant Race: 1",
-      24 -> "Co-applicant Race: 1",
-      29 -> "Applicant Sex",
-      30 -> "Co-applicant Sex",
-      32 -> "Type of Purchaser",
-      37 -> "HOEPA Status",
-      38 -> "Lien Status"
-    )
-
-    if (values.length != 40 && values.length != 39) {
-      return errors ::: List("Incorrect number of fields: " + values.length)
+  def toIntorFail(value: String, fieldName: String): ValidationNel[String, List[Int]] = {
+    Try(value.toInt) match {
+      case Failure(result) => s"$fieldName is not an Integer".failure.toValidationNel
+      case Success(result) => List(result).success
```

Comment:
Creating a wrapper object around a single `Int` seems like overkill, but I understand the need to not have the aggregation simply sum all the values.  Maybe there's a different way to aggregate these values without a need for a wrapper?